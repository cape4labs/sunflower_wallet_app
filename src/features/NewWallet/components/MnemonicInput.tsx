import { View, TextInput } from 'react-native';
import TextWithFont from '../../../shared/components/TextWithFont';

type MnemonicWordType = {
  idx: number;
  onChange: (text: string) => void;
  value: string;
};

// A component used for inputins words that are part of a mnemonic
export function MnemonicInput({ idx, onChange, value }: MnemonicWordType) {
  return (
    <View className="flex-row w-full px-2 my-1 p-2 bg-custom_border items-center rounded-lg">
      <TextWithFont customStyle="text-white">{idx}.</TextWithFont>
      <TextInput
        className="flex-1 h-full text-white rounded-md text-lg"
        onChangeText={onChange}
        autoCapitalize="none"
        value={value}
      />
    </View>
  );
}
