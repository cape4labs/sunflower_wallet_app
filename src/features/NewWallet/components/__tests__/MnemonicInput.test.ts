import { render } from '@testing-library/react-native';
import { MnemonicInput } from '../MnemonicInput';

it('MnemonicInput snapshot', () => {
  const tree = render(MnemonicInput({ idx: 0, onChange: () => {}, value: 'test' })).toJSON();
  expect(tree).toMatchSnapshot();
});
