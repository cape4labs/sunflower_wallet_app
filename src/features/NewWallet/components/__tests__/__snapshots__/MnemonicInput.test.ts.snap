// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicInput snapshot 1`] = `
<View
  className="flex-row w-full px-2 my-1 p-2 bg-custom_border items-center rounded-lg"
>
  <Text
    className="font-[spacegrotesk] font-normal text-white"
  >
    0
    .
  </Text>
  <TextInput
    autoCapitalize="none"
    className="flex-1 h-full text-white rounded-md text-lg"
    onChangeText={[Function]}
    value="test"
  />
</View>
`;
